libafl/src/stages/mutational.rs

```diff
@@ -78,14 +78,12 @@ where
 /// A Mutational stage is the stage in a fuzzing run that mutates inputs.
 /// Mutational stages will usually have a range of mutations that are
 /// being applied to the input one by one, between executions.
-pub trait MutationalStage<E, EM, I, M, Z>: Stage<E, EM, Z>
+pub trait MutationalStage<E, EM, I, M, S, Z>: Stage<E, EM, S, Z>
```

Comment:
The stage requirement should be dropped. The comment below w.r.t. removing the perform_mutational stage will shift out the generics from that function, so `MutationalStage<S>` should be the final definition. `M` should be changed to an associated type.

All trait bounds should be removed.